docs/README.md

```diff
@@ -66,11 +73,15 @@ $ sudo pip install 'sphinx<3.1.0' mkdocs numpy pydata_sphinx_theme ipython nbsph
 
 ### R API Documentation (Optional)
 
-If you'd like to generate R API documentation, you'll need to [install Pandoc](https://pandoc.org/installing.html)
-and install these libraries:
+If you'd like to generate R API documentation, you'll need to install these packages and libraries:
+
+```sh
+$ sudo apt install libssl-dev libcurl4-openssl-dev pandoc libfontconfig1-dev libharfbuzz-dev \
+          libfribidi-dev libfreetype6-dev libpng-dev libtiff5-dev libjpeg-dev libxml2-dev
+```
 
 ```sh
-$ sudo Rscript -e 'install.packages(c("knitr", "devtools", "testthat", "rmarkdown"), repos="https://cloud.r-project.org/")'
+$ sudo Rscript -e 'install.packages(c("curl", "knitr", "devtools", "testthat", "rmarkdown", "markdown", "e1071"), repos="https://cloud.r-project.org/")'
```

Comment:
Hm, `rmarkdown` depends on `markdown` IIRC. `rmarkdown` falls back to `markdown`. Was this required in your env?

Reply:
cc @huaxingao FYI who faced a similar problem before IIRC ..

Reply:
I think I finally managed to understand what's going on... 
So, I'm using [this Docker](https://github.com/khalidmammadov/spark/blob/e9cec4091b159e1c0c6c44a1fb816ca16a77e9f5/docs/Dockerfile) for the build. And tested with and without `markdown` package and it fails without and I couldn't understand how it succeeds in the [build and test CI](https://github.com/apache/spark/blob/94fd9c55c6a29208bbfe240bd2f3191c7df4c666/.github/workflows/build_and_test.yml#L537) phase. So, apparently it's installed on the base image (and others I am adding here) from @dongjoon-hyun 's [Docker image](https://hub.docker.com/layers/dongjoon/apache-spark-github-action-image/20220207/images/sha256-af09d172ff8e2cbd71df9a1bc5384a47578c4a4cc293786c539333cafaf4a7ce?context=explore) (BTW, where is the source of this Dockerfile kept?). So, some packages are "reinstalled" during Build and test and some not hence the confusion. 
Additionally, I tested building "without" `rmarkdown` and it succeeds.

Reply:
I had the same problem:  I tested with and without markdown package and it failed without.

Reply:
I re-tested that number of times in docker containers and it always fails if package is not installed. So, yes, in short `markdown` is required package.